Connected System Template (CST) Examples/Google Drive (OAuth,  Sending Files and Multiple Operations)/src/main/java/com/mycorp/helloworld/templates/utils/DiagnosticsFactory.java

```diff
@@ -0,0 +1,47 @@
+package com.mycorp.helloworld.templates.utils;
+
+import static com.mycorp.helloworld.templates.GoogleDriveSampleConnectedSystemTemplate.CLIENT_ID_KEY;
+import static com.mycorp.helloworld.templates.GoogleDriveSampleConnectedSystemTemplate.CLIENT_SECRET_KEY;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import com.appiancorp.connectedsystems.simplified.sdk.configuration.SimpleConfiguration;
+import com.google.api.services.drive.model.File;
+
+/**
+ * This class is an helper class that creates common fields of diagnostics objects for both Send File
+ * and Create Folder templates. The specific fields to the individual template will be added in
+ * {@link CreateFolderDiagnosticsFactory} and {@link SendFileDiagnosticsFactory}.
+ */
+public class DiagnosticsFactory {
+
+  protected DiagnosticsFactory() {}
+
+  protected Map<String,Object> getRequestDiagnostics(SimpleConfiguration connectedSystemConfiguration) {
+    Map<String,Object> requestDiagnostic = new HashMap<>();
+    //Request Diagnostic values will be shown on the Request tab on Appian Integration Designer Interface,
+    //which will be visible to designers. Only add to diagnostics values that you wish the designer to see.
+    String clientId = connectedSystemConfiguration.getValue(CLIENT_ID_KEY);
+    requestDiagnostic.put(CLIENT_ID_KEY, clientId);
+    //For sensitive values, mask it so that it won't be visible to designers
+    requestDiagnostic.put(CLIENT_SECRET_KEY, "******************");
+    return requestDiagnostic;
+  }
+
+  public Map<String,String> getResponseDiagnostics(File responseFile) {
+    Map<String,String> responseDiagnostic = new HashMap<>();
+    responseDiagnostic.put("Name", responseFile.getName());
+    responseDiagnostic.put("ID", responseFile.getId());
+    responseDiagnostic.put("MIME Type", responseFile.getMimeType());
+    return responseDiagnostic;
+  }
+
+  public Map<String,Object> getResult(File responseFile) {
```

Comment:
Strange to me that the diagnostic factory creates results

Reply:
i will move this outside to the Template classes. I am also trying to avoid creating too many classes in an example template.